exec/cnex/lib/time_posix.c

```diff
@@ -0,0 +1,40 @@
+#define _POSIX_C_SOURCE 200809L
+#include <stdlib.h>
+#include <sys/time.h>
+#include <time.h>
+#include <unistd.h>
+
+#include "cell.h"
+#include "exec.h"
+#include "number.h"
+#include "stack.h"
+
+extern Number NANOSECONDS_PER_SECOND;
+
+void time_sleep(TExecutor *exec)
+{
+    // Note: The number on the top of the stack is in SECONDS!
+    uint64_t ns = number_to_uint64(number_multiply(top(exec->stack)->number, NANOSECONDS_PER_SECOND)); pop(exec->stack);
+
+    // So we must get the correct NANOSECOND value to sleep the correct amount of time.
+    struct timespec ts;
+    if (ns > number_to_uint64(NANOSECONDS_PER_SECOND)) {
```

Comment:
Can you simply use a constant value here instead of converting from a constant to a Number and then back again?

Reply:
Yes.  That was my original idea, but since I already had that other Number defined, I tried to be smooth and use that, but it broke the OSX builds.   So...

Fixed.